Code/build.mak

```diff
@@ -42,20 +43,27 @@ AS_FLAGS=\
 
 CC_FLAGS_COMMON=\
 	-D__GCC__\
+	-DAP_DEFINED_LOCALLY\
+	-D__CORRECT_ISO_CPP_STRING_H_PROTO\
+	-D_PSP_FW_VERSION=500\
 	$(INCLUDE)\
 	-I. -c\
 	-include "ForcedInclude.h"\
 	-MMD    \
 	-Wall\
 	-fno-exceptions\
+	-Wno-deprecated-declarations\
 
 CC_FLAGS_DEBUG=\
 	-O0\
 	-g\
 	-D_DEBUG\
+	-fshort-wchar\
+	-ffast-math\
+	-fomit-frame-pointer\
 
 CC_FLAGS_RELEASE=\
-	-O3\
+	-Os\
```

Comment:
I don't think you want to do that - that would have quite an impact on codegen.
And definitely shouldn't be required..

Reply:
currently O3 produces worse performing code on mips-allegrex as is the case for most platforms over O2 or Os

Reply:
While that may be true in the general case, it's always good to back up such assumptions with real-word data.
I disabled the rendering (making it completely CPU bound) and calculated simulation tick to wall-clock tick ratio for the entire run and compared `-Os`,`-O2`,`-O3`.

`-Os` is slower across the board. GCC 9.3.0 fairs slightly better than GCC 4.1.0 ; **61.1%** (930) vs **63.7%** (410).
`-O2` and `-O3` is practically the same for GCC 9.3.0 (**56.7%**), while for 4.1.0 it varies more per scene, but overall `-O3` is better than `-O2`; **56.9%** (-O3) vs **57.3%** (-O2).
(lower % is better obviously)

Conclusion : 
- Changing to `-Os` would create significantly worse code than `-O2`/`-O3`.
- `-O2` would be acceptable when using GCC 9.3.0 but would cause a regression when using GCC 4.1.0.